services/viva-ms/lambdas/checkCompletionsStatus.js

```diff
@@ -0,0 +1,59 @@
+/* eslint-disable no-console */
+import to from 'await-to-js';
+
+import log from '../../../libs/logs';
+
+import putVivaMsEvent from '../helpers/putVivaMsEvent';
+import vivaAdapter from '../helpers/vivaAdapterRequestClient';
+import validateApplicationStatus from '../helpers/validateApplicationStatus';
+
+export async function main(event, context) {
+  const clientUser = event.detail;
+
+  const [applicationStatusError, applicationStatusList] = await to(
+    vivaAdapter.application.status(clientUser.personalNumber)
+  );
+  if (applicationStatusError) {
+    log.error(
+      'Viva application status failed',
+      context.awsRequestId,
+      'service-viva-ms-checkCompletionsStatus-001',
+      applicationStatusError
+    );
+    return false;
+  }
+
+  /**
+   * The Combination of Status Codes 64, 128, 256, 512
+   * determines if a VIVA Application Workflow requires completion
+   * 64 - Completion is required,
+   * 128 - Case exsits in VIVA
+   * 256 - An active e-application is activated in VIVA
+   * 512 - Application allows e-application
+   */
+  const completionStatusCodes = [64, 128, 256, 512];
```

Comment:
Im not fond of having "dangling" values like 64 .. etc. Can this be put in a constant instead like:

```js
// in constant file
const VIVA_STATUS_CODE = {
  COMPLETION_REQUIRED: 64,
  CASE_EXIST: 128,
  APPLICATION_ACTIVATED: 256,
  E_APPLICATION_ALLOWED: 512
}
```
```js
// in checkCompletionsStatus.js
const { VIVA_STATUS_CODE } from "../some_constant_file";

const completionStatusCodes = Object.values(VIVA_STATUS_CODE);


export async function main(event, context) { 
  ...

  const hasInvalidStatusCodeCombination = !validateApplicationStatus(applicationStatusList, completionStatusCodes);

if (hasInvalidStatusCodeCombination) { ... }

}

...

log.info(
      `No Viva completion status(${VIVA_STATUS_CODE.COMPLETION_REQUIRED}) found`,
      context.awsRequestId,
      'service-viva-ms-checkCompletionsStatus-002',
      applicationStatusList
    );
```